offchain/dispatcher/src/dispatcher.rs

```diff
@@ -57,19 +58,15 @@ pub async fn start(
             .into(),
     );
 
-    let initial_state = RollupsInitialState {
-        history_address: config
-            .blockchain_config
-            .history_address
-            .clone()
-            .into_inner()
-            .into(),
-        input_box_address: config
-            .blockchain_config
-            .input_box_address
-            .clone()
-            .into_inner()
-            .into(),
+    let initial_state = InputBoxInitialState {
+        input_box_address: Arc::new(
+            config
+                .blockchain_config
+                .input_box_address
+                .clone()
+                .into_inner()
+                .into(),
```

Comment:
These two lines looked strange to me, but I think I get it now. There are two definitions for `Address`: one from `ethers.rs` and another from our broker. I imagine to make the rest of the node not depend on `ethers.rs`.